### PR TITLE
feat(auth): forward host auth config to auth handler plugins

### PR DESCRIPTION
**Pull request by kcloutie**
- inject per-handler auth settings (github, entra, gcp) from app config into plugin ProviderConfig.Settings during registration
- clone Settings map before mutation to avoid shared-state bugs
- auto-resolve official auth handlers in root command for direct CLI commands that bypass the solution prepare pipeline
- bump scafctl-plugin-sdk to v0.5.1
- add tests for injectAuthHandlerSettings covering all handlers and nil/unknown edge cases

### CODE REVIEW

File: pkg/cmd/scafctl/root.go [hunk 1]
```diff
@@ -586,6 +587,38 @@ func Root(opts *RootOptions) *cobra.Command {
...
+			if !cfg.Settings.DisableOfficialAuthHandlers {
+				var cooldownDuration time.Duration
+				if cfg.Plugins.FetchCooldown != "" {
+					if d, parseErr := time.ParseDuration(cfg.Plugins.FetchCooldown); parseErr == nil {
+						cooldownDuration = d
+					} else {
+						lgr.V(1).Info("invalid plugins.fetchCooldown duration, using default", "value", cfg.Plugins.FetchCooldown, "error", parseErr)
+					}
+				}
+				cooldown := plugin.NewFetchCooldown(paths.PluginCacheDir(), cooldownDuration)
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/cmd/scafctl/root.go]: Fixed — switched to `settings.PluginCacheDirFor(binaryName)` so cooldown markers are isolated per embedding binary. Good catch, thanks!

File: pkg/cmd/scafctl/root.go [hunk 2]
```diff
@@ -586,6 +587,38 @@ func Root(opts *RootOptions) *cobra.Command {
...
+					Quiet:      cliParams.IsQuiet,
+					NoColor:    cliParams.NoColor,
+					BinaryName: binaryName,
+				}
+				hostDeps := plugin.HostDepsFromAuthRegistry(authRegistry)
+				if hostDeps != nil && secretErr == nil {
+					hostDeps.SecretStore = sharedSecretStore
+				}
+				clientOpts := []plugin.ClientOption{plugin.WithHostDeps(hostDeps)}
+				officialAuthClients, resolveErr := solprepare.ResolveOfficialAuthHandlers(ctx, authRegistry, cooldown, pluginCfg, clientOpts...)
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/cmd/scafctl/root.go:622]: Fixed — moved the signature-policy wiring (`plugin.WithSignaturePolicy`) before `ResolveOfficialAuthHandlers` so auto-fetched auth handler plugins now respect the embedder's `PluginSignaturePolicy`. Good catch, thanks!

File: pkg/cmd/scafctl/root.go [hunk 3]
```diff
@@ -586,6 +587,43 @@ func Root(opts *RootOptions) *cobra.Command {
...
+				}
+				hostDeps := plugin.HostDepsFromAuthRegistry(authRegistry)
+				if hostDeps != nil && secretErr == nil {
+					hostDeps.SecretStore = sharedSecretStore
+				}
+				// Wire plugin signature policy before resolving official auth
+				// handlers so auto-fetched plugins respect the embedder's policy.
+				if opts.PluginSignaturePolicy != nil {
+					ctx = plugin.WithSignaturePolicy(ctx, opts.PluginSignaturePolicy)
+				}
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/cmd/scafctl/root.go]: Fixed — moved `plugin.WithSignaturePolicy` unconditionally before both the auth-handler and provider blocks, so the embedder's policy is always applied regardless of `DisableOfficialAuthHandlers`. Good catch, thanks!

[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/cmd/scafctl/root.go]: Fixed — moved plugin.WithSignaturePolicy unconditionally before both the auth-handler and provider blocks, so the embedder policy is always applied regardless of DisableOfficialAuthHandlers. Good catch, thanks!

File: pkg/plugin/wrapper_auth.go [hunk 4]
```diff
@@ -229,7 +231,9 @@ func configureAndRegisterAuthHandlers(ctx context.Context, registry *auth.Regist
 	for _, info := range handlers {
 		if cfg != nil {
 			hostCfg := *cfg
+			hostCfg.Settings = maps.Clone(cfg.Settings)
 			hostCfg.HostServiceID = client.HostServiceID()
+			injectAuthHandlerSettings(ctx, info.Name, &hostCfg)
 			if cfgErr := client.ConfigureAuthHandler(ctx, info.Name, hostCfg); cfgErr != nil {
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/plugin/wrapper_auth.go]: Fixed — reordered configureAndRegisterAuthHandlers to pre-check registry.Has(name) and skip entirely on collision. Registration now happens before ConfigureAuthHandler, so secrets are never sent to a plugin that will not be used. Added a test (DuplicateSkipsConfig) verifying ConfigureAuthHandler is not called on name collision. Good catch, thanks!
